control-plane/cmd/kafka-controller/kodata/dataplane/config/broker/100-config-kafka-broker-data-plane.yaml

```diff
@@ -0,0 +1,179 @@
+# Copyright 2020 The Knative Authors
```

Comment:
Receiver, dispatcher and this CM are exactly the same as the other in `data-plane/config/...` (the CM is actually already out of date ...), can we just add to `./hack/update-codegen.sh` `cp <dispatcher_path> <ko_data_dispatcher_path>` so that we don't need to keep them in sync manually?

Other thought, since we need to support customizations, I believe creating manifests on the fly based on the ones in the system namespace would provide the customizations we need and it would also avoid injecting the images into the controller deployment.

Example:
- get receiver, dispatcher and CM from `system.Namespace()`
- get SA and role from the kodata filesystem
- create manifests with all the transformations that we're doing

In this way, if I update, for example, the receiver with an addition env variable, I get that propagated to all other receivers in my cluster.

Example patch for CM:
```
diff --git a/control-plane/pkg/reconciler/broker/namespaced_broker.go b/control-plane/pkg/reconciler/broker/namespaced_broker.go
index 7fa566f1..32bf0354 100644
--- a/control-plane/pkg/reconciler/broker/namespaced_broker.go
+++ b/control-plane/pkg/reconciler/broker/namespaced_broker.go
@@ -27,18 +27,20 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/scheme"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/util/retry"
 	"k8s.io/utils/pointer"
+	eventing "knative.dev/eventing/pkg/apis/eventing/v1"
+	"knative.dev/pkg/reconciler"
+	"knative.dev/pkg/resolver"
+
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/config"
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/kafka"
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/prober"
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/base"
-	eventing "knative.dev/eventing/pkg/apis/eventing/v1"
-	"knative.dev/pkg/reconciler"
-	"knative.dev/pkg/resolver"
 )
 
 // TODO: 1.
@@ -164,6 +166,11 @@ func (r *NamespacedReconciler) createReconcilerForBrokerInstance(broker *eventin
 func (r *NamespacedReconciler) getManifest(broker *eventing.Broker) (mf.Manifest, error) {
 	manifest := r.BaseDataPlaneManifest
 
+	cmManifest := r.dataPlaneConfigMapManifest()
+	// TODO do the same for receiver and dispatcher deployments/statefulsets
+
+	manifest.Append(cmManifest)
+
 	manifest, err := manifest.Transform(mf.InjectNamespace(broker.Namespace))
 	if err != nil {
 		return mf.Manifest{}, fmt.Errorf("unable to transform base dataplane manifest with namespace injection. namespace: %s, error: %v", broker.Namespace, err)
@@ -172,6 +179,15 @@ func (r *NamespacedReconciler) getManifest(broker *eventing.Broker) (mf.Manifest
 	return manifest.Transform(appendNewOwnerRefsToPersisted(manifest.Client, broker))
 }
 
+// TODO handle errors
+func (r *NamespacedReconciler) dataPlaneConfigMapManifest() mf.Manifest {
+	cm, _ := r.ConfigMapLister.ConfigMaps(r.Env.DataPlaneConfigMapNamespace).Get(r.Env.DataPlaneConfigMapName)
+	cmUnstr, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(cm)
+	u := unstructured.Unstructured{Object: cmUnstr}
+	cmManifest, _ := mf.ManifestFrom(mf.Slice{u})
+	return cmManifest
+}
+
 func appendNewOwnerRefsToPersisted(client mf.Client, broker *eventing.Broker) mf.Transformer {
 	return func(resource *unstructured.Unstructured) error {
 		existingRefs, err := getPersistedOwnerRefs(client, resource)
```

Reply:
The same approach can be used for config-tracing, logging, etc (that we're removing)

Reply:
IMO let's discuss this separately.

Reply:
Ok

Reply:
but we need to update manifests ...

Reply:
I am gonna merge this in a disabled way. Things work now w/o configmaps mounted.
Manifests can be updated later when we're ready with the whole thing.

Or, do you mean something else?

Reply:
> I am gonna merge this in a disabled way. Things work now w/o configmaps mounted.
Manifests can be updated later when we're ready with the whole thing.

This CM config-kafka-broker-data-plane is still mounted in the new data plane and it's different from the one in `data-plane/config/...`, feel free to ignore this comment for now but the current situation is not optimal, given the CM config-kafka-broker-data-plane is already out of date ...